forte/data/data_store.py

```diff
@@ -759,27 +772,41 @@ def _add_entry_raw(
         Returns:
             ``tid`` of the entry.
         """
-        if entry_type in (Annotation, AudioAnnotation):
+        if any(
+            self._is_subclass(type_name, cls)
+            for cls in (Annotation, AudioAnnotation)
+        ):
+            begin_idx = self._type_attributes[type_name][
+                constants.TYPE_ATTR_KEY
+            ]["begin"]
+            end_idx = self._type_attributes[type_name][constants.TYPE_ATTR_KEY][
+                "end"
+            ]
+
             sorting_fn = lambda s: (
-                s[constants.BEGIN_INDEX],
-                s[constants.END_INDEX],
+                s[begin_idx],
+                s[end_idx],
             )
             try:
                 self.__elements[type_name].add(entry)
             except KeyError:
                 self.__elements[type_name] = SortedList(key=sorting_fn)
                 self.__elements[type_name].add(entry)
-        elif entry_type in [
-            Link,
-            Group,
-            Generics,
-            ImageAnnotation,
-            Grids,
-            Payload,
-            MultiPackLink,
-            MultiPackGroup,
-            MultiPackGeneric,
-        ]:
+
+        elif any(
+            self._is_subclass(type_name, cls)
+            for cls in [
+                Link,
+                Group,
+                Generics,
+                ImageAnnotation,
+                Grids,
+                Payload,
+                MultiPackLink,
+                MultiPackGroup,
+                MultiPackGeneric,
+            ]
+        ):
```

Comment:
Does this mean that we have to update this list for every new `top` type we created? Can we check whether `cls` is `entry`?

Reply:
We can't just check for `Entry` because just above that there is a check for `Annotation` and `AudioAnnotation` as well.

Reply:
why would the check for `Annotation` affect here? We first check for `Annotation`, if not, then we check for the other `Entry`, no?

Reply:
I'll try that